tools/override-generator/main.go

```diff
@@ -0,0 +1,286 @@
+	for _, r := range repos {
+		if r.Kind() == "go_repository" {
+			repoOverrides := goRepositoryToOverrideSet(r)
+			outputOverrides = append(outputOverrides, setToOverridesSlice(repoOverrides)...)
+		}
+	}
+
+	f := rule.EmptyFile("generated", "")
+	for _, o := range outputOverrides {
+		o.Insert(f)
+	}
+
+	if err := f.Save(a.outputFile); err != nil {
+		return fmt.Errorf("error saving file: %w", err)
+	}
+
+	return nil
+}
+
+func goRepositoryToOverrideSet(r *rule.Rule) overrideSet {
+	// each repo has its own override set, and can't have multiple
+	// duplicate overrides. This set is created to be populated and read
+	set := make(overrideSet)
+
+	importPath := r.AttrString("importpath")
+	for attr, keys := range _mapAttrToOverride {
+		attrValue := r.Attr(attr)
+		if attrValue == nil || attr == _buildFileProtoModeAttr {
+			continue
+		}
+
+		override := rule.NewRule(keys.kind, "")
+		if o, ok := set[keys.kind]; ok {
+			override = o
+		}
+		override.SetAttr("path", importPath)
+
+		for _, a := range keys.attrs {
+			val := r.Attr(a)
+
+			// Special case for certain renamed attributes like "build_directives"
+			// attribute to convert to "directives" attribute.
+			if k, ok := _attrOverrideKeys[a]; ok {
+				a = k
+			}
+
+			if def, ok := _defaultValues[a]; def == r.AttrString(a) && ok {
+				continue
+			}
+
+			if val != nil {
+				switch v := val.(type) {
+				case *build.StringExpr:
+					override.SetAttr(a, v)
+				case *build.ListExpr:
+					// Special case for "patch_args" attribute to convert to
+					// "patch_strip" attribute.
+					if a == _patchArgsAttr {
+						setPatchArgs(v, override)
+						continue
+					}
+					override.SetAttr(a, v)
+				}
+			}
+		}
+		set[keys.kind] = override
+	}
+
+	// Since "build_file_proto_mode" is added to the "directives", we need
+	// to run it after the fact to make sure that "directives" is set.
+	applyBuildFileProtoMode(r, set)
+	return set
+}
+
+func applyBuildFileProtoMode(r *rule.Rule, set overrideSet) {
+	a := _buildFileProtoModeAttr
+	if r.Attr(a) == nil {
+		return
+	}
+
+	if def, ok := _defaultValues[a]; def == r.AttrString(a) && ok {
+		return
+	}
+
+	directive := "gazelle:proto " + r.AttrString(_buildFileProtoModeAttr)
+	kind := _gazelleOverride
+	override := rule.NewRule(kind, "")
+	if o, ok := set[kind]; ok {
+		override = o
+	}
+	if list := override.Attr(_directivesAttr); list != nil {
+		if listExpr, ok := list.(*build.ListExpr); ok {
+			listExpr.List = append(listExpr.List, &build.StringExpr{Value: directive})
+		}
+		return
+	}
+	override.SetAttr(_directivesAttr, []string{directive})
+	set[kind] = override
+}
+
+func setPatchArgs(v *build.ListExpr, override *rule.Rule) {
+	for _, li := range v.List {
+		if liStr, ok := li.(*build.StringExpr); ok {
+			if strings.HasPrefix(liStr.Value, "-p") {
+				numStr := strings.TrimPrefix(liStr.Value, "-p")
+				if num, err := strconv.Atoi(numStr); err == nil {
+					override.SetAttr("patch_strip", num)
+					return
+				}
+			}
+		}
```

Comment:
```suggestion

	patch_args = override.AttrStrings("patch_args")
	for _, arg := range patch_args {
		if !strings.HasPrefix(arg, "-p") {
			continue
		}
		numStr := strings.TrimPrefix(arg, "-p")
		if num, err := strconv.Atoi(numStr); err == nil {
			override.SetAttr("patch_strip", num)
			return
		}	
	}
```
simpler and less nesting

Reply:
Yep! I actually need to get the `AttrStrings` outside of this function but this works nicely.